unblob/pool.py

```diff
@@ -67,13 +89,16 @@ def __init__(
             )
             for _ in range(process_num)
         ]
+        self._should_stop_processing = False
+        _register_signal_handler()
```

Comment:
I am really not fond of registering signal handlers from other places than the vicinity of the application's entrypoint. Also, libraries should never mess with the process's signal handlers from which they are called from.

Reply:
Totally agree!

In offline, we talked about how to handle this. Our (newest) idea is to save the original signal handlers for the signals we want to create our own, then when we handled the signals and exited revert back the signal handlers to the original.
This way we can (hopefully) be sure that does not mess with the caller's signal setup.